photutils/psf/photometry.py

```diff
@@ -274,10 +278,17 @@ def do_photometry(self, image, positions=None):
                                   sources['ycentroid'],
                                   sources['aperture_flux']])
 
+        self._pars_to_set, self._pars_to_output = self._define_fit_param_names()
```

Comment:
I'm finding it a bit confusing to understand why `_pars_to_set` and `_pars_to_output` are set the way they are.  I suggest one of a few alternatives:

1. Make `_define_fit_param_names` *internally* do the setting, instead of it returning values that you then immediately set.  Then the call can just be ``self._define_fit_param_names()`` (you can leave the text in the docstrong for `_define_fit_param_names` about what these values are so that it's documented somewhere.  
2. Have `_pars_to_set` and `_pars_to_output` be only *local* variables in `do_photometry`.  I guess you'll then need to then change `nstar` to call ``self._define_fit_param_names()`` at the top, separately.
3. Have  `_pars_to_set` and `_pars_to_output` get set *when the model attribute is set*, rather than when `do_photometry` is called.  I think this is a simple as adding the snippet I've but at the bottom of this post, and removing this line. This makes more sense because then if someone, say, wants to subclass this, they don't need to remember to call `define_fit_param_names` at the top of `do_photometry` to replicate all the functionality.


```
@property
def model(self):
    return self._model
@model.setter
def model(self, val):
    self._model = val
    self._define_fit_param_names()
```